lib/sources.nix

```diff
@@ -270,6 +270,19 @@ let
       outPath = builtins.path { inherit filter name; path = origSrc; };
     };
 
+  predicateFilter = { name ? "source", src, predicate }:
```

Comment:
I don't think this is a move in the right direction. By repeating each feature in the argument list, we're creating the square of the simpler api proposed in
 - https://github.com/NixOS/nixpkgs/pull/112083

I strongly suggest going with the two-argument `filter` and `setName` functions. They do not need extensibility because they have a single, well-defined responsibility. Any new responsibilities can be covered by new functions instead. Let's keep it simple.